pybamm/geometry/standard_spatial_vars.py

```diff
@@ -182,3 +182,65 @@
     },
     coord_sys="cartesian",
 )
+
+# Spatial variables for the core and shell phases in the core-shell model
+# for PE phase transition caused degradation
+# the core domain (0, s) and shell domain (s, R) have been mapped to (0, R_typ)
+# core variable  r_co = r / s * R_typ, r goes from 0 to s
+# shell variable r_sh = (r - s) / (R - s) * R_typ, r goes from s to R
+r_co = pybamm.SpatialVariable(
+    "r_co",
+    domain=["positive core"],
+    auxiliary_domains={
+        "secondary": "positive electrode",
+        "tertiary": "current collector",
+    },
+    coord_sys="spherical polar",
+)
+r_sh = pybamm.SpatialVariable(
+    "r_sh",
+    domain=["positive shell"],
+    auxiliary_domains={
+        "secondary": "positive electrode",
+        "tertiary": "current collector",
+    },
+    # note the coordinate system is different from that of r_co
```

Comment:
if you're doing a transformation anyway, why not pose both eqns on a catersian domain [0, 1], then do the mapping later? assume this is to make plotting easier later so everything lives on [0, R]?